back-end/src/common/gateways/retrospective.global.gateway.ts

```diff
@@ -101,4 +101,82 @@ export class RetrospectiveGateway
 		};
 		client.to(client.roomId).emit('peer-mouse-moved', returnData);
 	}
+
+
+	@SubscribeMessage('end-currentRetro')
+	async endCurrentRetro(client: AuthSocket) {
+		const endedDate = new Date();
+		client.to(client.roomId).emit('end-currentRetro', endedDate);
+
+		const query = { slug: client.roomId };
+		const update = {$set: { endedAt: endedDate, isRetroEnded: true}}
+
+		await this.retrospectivesRepository.updateOneRetrospective(query, update);
```

Comment:
If i emit this event, the retro end early.
Fix it.

Reply:
Cet event est reçu que lorsque la rétro arrive à 0 donc, je ne vois pas comment on peut l'emit avant ?

Reply:
Not fixed, but merging.